src/main/data_structure/stack.py

```diff
@@ -0,0 +1,41 @@
+class Stack:
+
+    # Constructor to initiate stack
+    def __init__(self, stack=None):
+        if stack is None:
+            stack = []
+        self.stack = stack
+
+    # this returns the top item of stack
+    def top(self):
```

Comment:
prev code was okay. we just need to throw an exception like

raise Exception('You are trying to top an empty Stack')
we don't need try catch here